src/NHibernate/Cfg/Configuration.cs

```diff
@@ -238,18 +236,18 @@ public bool HasNonIdentifierPropertyNamedId(string className)
 			}
 
 			public Dialect.Dialect Dialect =>
-				NHibernate.Dialect.Dialect.GetDialect(configuration.Properties);
+				throw new InvalidOperationException("The dialect is not ready at this stage of the configuration.");
```

Comment:
To prevent further misuses of this property - throw IOE.

Reply:
I think we don't need this exception anymore. Now misuse in our code is prevented by `GetMapping` obsoletion.

Reply:
Reverted